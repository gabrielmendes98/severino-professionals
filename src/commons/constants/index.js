export const SEVERINO_CLIENTS_WEBSITE = 'https://www.google.com'; // trocar quando fizer deploy do outro site

export const SEVERINO_EMAIL = 'severino@gmail.com';

export const ERRORS = {
  DEFAULT: 'Ocorreu um erro inesperado, tente novamente.',
};

export const SUCCESS_OPERATION_MESSAGE = 'Operação realizada com sucesso!';

export const DATE_LOCALE = 'pt-br';
