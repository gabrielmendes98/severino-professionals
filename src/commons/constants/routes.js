const PAGE_URL = {
  HOME: '/',
  ABOUT_US: '/about-us',
  TERMS_AND_CONDITIONS: '/terms-and-conditions',
  SIGN_UP: '/sign-up',
  PROFILE: '/profile',
  LOGIN: '/login',
};

export default PAGE_URL;
