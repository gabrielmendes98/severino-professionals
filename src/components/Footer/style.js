import styled from 'styled-components';
import AppBrand from 'components/AppBrand';

export const FooterContainer = styled.footer`
  display: flex;
  background-color: ${({ theme }) => theme.palette.common.white};
  border-top: 1px solid ${({ theme }) => theme.palette.grey.lighter};
  box-shadow: 0px 1px ${({ theme }) => theme.spacing(0.5)}px rgba(0, 0, 0, 0.25);

  padding-top: ${({ theme }) => theme.spacing(5)}px;
  padding-bottom: ${({ theme }) => theme.spacing(5)}px;
`;

export const StyledAppBrand = styled(AppBrand)`
  text-align: center;
`;
