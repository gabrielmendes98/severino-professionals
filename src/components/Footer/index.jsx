import { Container, IconButton } from '@material-ui/core';
import { Link } from 'react-router-dom';
import FacebookIcon from '@material-ui/icons/Facebook';
import TwitterIcon from '@material-ui/icons/Twitter';
import InstagramIcon from '@material-ui/icons/Instagram';
import { SEVERINO_CLIENTS_WEBSITE, SEVERINO_EMAIL } from 'commons/constants';
import PAGE_URL from 'commons/constants/routes';
import Text from 'components/Text';
import { Grid } from 'components/Styled';
import { FooterContainer, StyledAppBrand } from './style';

const Footer = () => (
  <Grid container margin={{ top: 10 }} component={FooterContainer}>
    <Container maxWidth="xl" style={{ display: 'flex' }}>
      <Grid container item sm={4} justifyContent="center">
        <Grid container item xs={12} justifyContent="center">
          <StyledAppBrand />
        </Grid>
        <Grid item xs={1}>
          <IconButton aria-label="facebook">
            <FacebookIcon />
          </IconButton>
        </Grid>
        <Grid item xs={1}>
          <IconButton aria-label="twitter">
            <TwitterIcon />
          </IconButton>
        </Grid>
        <Grid item xs={1}>
          <IconButton aria-label="instagram">
            <InstagramIcon />
          </IconButton>
        </Grid>
      </Grid>
      <Grid container item sm={4} alignContent="center" direction="column">
        <Text color="secondary" weight="bold" size={1.1} margin={{ bottom: 2 }}>
          Ajuda &amp; Dúvidas
        </Text>
        <Text margin={{ bottom: 2 }}>
          <a href={SEVERINO_CLIENTS_WEBSITE} target="_blank" rel="noreferrer">
            Site dos clientes
          </a>
        </Text>
        <Text margin={{ bottom: 2 }}>
          <Link to={PAGE_URL.ABOUT_US}>Sobre nós</Link>
        </Text>
        <Text>
          <Link to={PAGE_URL.TERMS_AND_CONDITIONS}>Termos e condições</Link>
        </Text>
      </Grid>
      <Grid container item sm={4} alignContent="center" direction="column">
        <Text color="secondary" weight="bold" size={1.1} margin={{ bottom: 2 }}>
          Contatos
        </Text>
        <Text>
          <a href={`mailto:${SEVERINO_EMAIL}`}>severino@gmail.com</a>
        </Text>
      </Grid>
    </Container>
  </Grid>
);

export default Footer;
