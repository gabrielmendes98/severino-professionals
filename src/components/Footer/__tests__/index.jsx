import {
  toMatchSnapshot,
  renderWithRouter,
  userEvent,
  screen,
} from 'test-utils';
import PAGE_URL from 'commons/constants/routes';
import { SEVERINO_CLIENTS_WEBSITE, SEVERINO_EMAIL } from 'commons/constants';
import Footer from '..';

it('should match snapshot', () => {
  toMatchSnapshot(<Footer />, { useRouter: true });
});

describe('should redirect to:', () => {
  beforeEach(() => {
    renderWithRouter(<Footer />);
  });

  it('about-us', () => {
    userEvent.click(screen.getByRole('link', { name: /sobre nós/i }));
    expect(window.location.pathname).toBe(PAGE_URL.ABOUT_US);
  });

  it('terms-and-condition', () => {
    userEvent.click(screen.getByRole('link', { name: /termos e condições/i }));
    expect(window.location.pathname).toBe(PAGE_URL.TERMS_AND_CONDITIONS);
  });
});

describe('link should have attribute:', () => {
  beforeEach(() => {
    renderWithRouter(<Footer />);
  });

  it('client website', () => {
    expect(
      screen.getByRole('link', { name: /site dos clientes/i }),
    ).toHaveAttribute('href', SEVERINO_CLIENTS_WEBSITE);
  });

  it('email', () => {
    expect(screen.getByRole('link', { name: SEVERINO_EMAIL })).toHaveAttribute(
      'href',
      `mailto:${SEVERINO_EMAIL}`,
    );
  });
});
