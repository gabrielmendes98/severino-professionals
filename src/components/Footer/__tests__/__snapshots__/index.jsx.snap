// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  .c4 {
  margin-top: 0px;
  margin-left: 0px;
  margin-right: 0px;
  margin-bottom: 16px;
  color: #3700B3;
  font-weight: bold;
  font-size: 1.1rem;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  font-family: Roboto;
}

.c5 {
  margin-top: 0px;
  margin-left: 0px;
  margin-right: 0px;
  margin-bottom: 16px;
  color: #363636;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  font-family: Roboto;
}

.c6 {
  margin-top: 0px;
  margin-left: 0px;
  margin-right: 0px;
  margin-bottom: 0px;
  color: #363636;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  font-family: Roboto;
}

.c1 {
  margin-top: 80px;
  margin-left: 0px;
  margin-right: 0px;
  margin-bottom: 0px;
}

.c2 {
  margin: 0;
  cursor: pointer;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #FFFFFF;
  border-top: 1px solid #DDDDDD;
  box-shadow: 0px 1px 4px rgba(0,0,0,0.25);
  padding-top: 40px;
  padding-bottom: 40px;
}

.c3 {
  text-align: center;
}

<footer
    class="c0 MuiGrid-root c1 MuiGrid-container"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthXl"
      style="display: flex;"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-content-xs-center MuiGrid-grid-sm-4"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-content-xs-center MuiGrid-grid-xs-12"
        >
          <img
            alt="Severino"
            class="c2 c3"
            src="[object Object]"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
        >
          <button
            aria-label="facebook"
            class="MuiButtonBase-root MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M5 3h14a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2m13 2h-2.5A3.5 3.5 0 0 0 12 8.5V11h-2v3h2v7h3v-7h3v-3h-3V9a1 1 0 0 1 1-1h2V5z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
        >
          <button
            aria-label="twitter"
            class="MuiButtonBase-root MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M22.46 6c-.77.35-1.6.58-2.46.69.88-.53 1.56-1.37 1.88-2.38-.83.5-1.75.85-2.72 1.05C18.37 4.5 17.26 4 16 4c-2.35 0-4.27 1.92-4.27 4.29 0 .34.04.67.11.98C8.28 9.09 5.11 7.38 3 4.79c-.37.63-.58 1.37-.58 2.15 0 1.49.75 2.81 1.91 3.56-.71 0-1.37-.2-1.95-.5v.03c0 2.08 1.48 3.82 3.44 4.21a4.22 4.22 0 0 1-1.93.07 4.28 4.28 0 0 0 4 2.98 8.521 8.521 0 0 1-5.33 1.84c-.34 0-.68-.02-1.02-.06C3.44 20.29 5.7 21 8.12 21 16 21 20.33 14.46 20.33 8.79c0-.19 0-.37-.01-.56.84-.6 1.56-1.36 2.14-2.23z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
        >
          <button
            aria-label="instagram"
            class="MuiButtonBase-root MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7.8 2h8.4C19.4 2 22 4.6 22 7.8v8.4a5.8 5.8 0 0 1-5.8 5.8H7.8C4.6 22 2 19.4 2 16.2V7.8A5.8 5.8 0 0 1 7.8 2m-.2 2A3.6 3.6 0 0 0 4 7.6v8.8C4 18.39 5.61 20 7.6 20h8.8a3.6 3.6 0 0 0 3.6-3.6V7.6C20 5.61 18.39 4 16.4 4H7.6m9.65 1.5a1.25 1.25 0 0 1 1.25 1.25A1.25 1.25 0 0 1 17.25 8 1.25 1.25 0 0 1 16 6.75a1.25 1.25 0 0 1 1.25-1.25M12 7a5 5 0 0 1 5 5 5 5 0 0 1-5 5 5 5 0 0 1-5-5 5 5 0 0 1 5-5m0 2a3 3 0 0 0-3 3 3 3 0 0 0 3 3 3 3 0 0 0 3-3 3 3 0 0 0-3-3z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-align-content-xs-center MuiGrid-grid-sm-4"
      >
        <p
          class="MuiTypography-root c4 MuiTypography-body1"
          margin="[object Object]"
          size="1.1"
          textcolor="secondary"
          transform=""
          weight="bold"
        >
          Ajuda & Dúvidas
        </p>
        <p
          class="MuiTypography-root c5 MuiTypography-body1"
          margin="[object Object]"
          transform=""
        >
          <a
            href="https://www.google.com"
            rel="noreferrer"
            target="_blank"
          >
            Site dos clientes
          </a>
        </p>
        <p
          class="MuiTypography-root c5 MuiTypography-body1"
          margin="[object Object]"
          transform=""
        >
          <a
            href="/about-us"
          >
            Sobre nós
          </a>
        </p>
        <p
          class="MuiTypography-root c6 MuiTypography-body1"
          transform=""
        >
          <a
            href="/terms-and-conditions"
          >
            Termos e condições
          </a>
        </p>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-align-content-xs-center MuiGrid-grid-sm-4"
      >
        <p
          class="MuiTypography-root c4 MuiTypography-body1"
          margin="[object Object]"
          size="1.1"
          textcolor="secondary"
          transform=""
          weight="bold"
        >
          Contatos
        </p>
        <p
          class="MuiTypography-root c6 MuiTypography-body1"
          transform=""
        >
          <a
            href="mailto:severino@gmail.com"
          >
            severino@gmail.com
          </a>
        </p>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
